react/features/e2ee/middleware.js

```diff
@@ -0,0 +1,43 @@
+// @flow
+
+import { getCurrentConference } from '../base/conference';
+import { MiddlewareRegistry, StateListenerRegistry } from '../base/redux';
+
+import { SET_E2EE_KEY } from './actionTypes';
+import { setE2EEKey } from './actions';
+import logger from './logger';
+
+/**
+ * Middleware that captures actions related to E2EE.
+ *
+ * @param {Store} store - The redux store.
+ * @returns {Function}
+ */
+MiddlewareRegistry.register(({ getState }) => next => action => {
+    switch (action.type) {
+    case SET_E2EE_KEY: {
+        const conference = getCurrentConference(getState);
+
+        if (conference) {
+            logger.debug(`New E2EE key: ${action.key}`);
+            conference.setE2EEKey(action.key);
+        }
+
+        break;
+    }
+    }
+
+    return next(action);
+});
+
+/**
+ * Set up state change listener to perform maintenance tasks when the conference
```

Comment:
Why not do this in a middleware? 

Currently isn't this going to be executed not when you leave the conference but when you actually create a new one? 

What is the scenario that you are trying to handle here?

Reply:
Yees, we you switch from one to another. I'm making sure the key is cleaned up. A middleware is much messier because you need to account for all failure types.

Reply:
OK. I see your point. But the still currently you'll keep the key from the time that you've left the conference until you create another one. Which doesn't seem very clean!